(window.webpackJsonp=window.webpackJsonp||[]).push([[10],{78:function(e,t,n){"use strict";n.r(t),n.d(t,"frontMatter",(function(){return o})),n.d(t,"metadata",(function(){return l})),n.d(t,"toc",(function(){return c})),n.d(t,"default",(function(){return u}));var r=n(3),a=n(7),i=(n(0),n(87)),o={id:"scn_dataset",title:"scNavigator dataset",slug:"/"},l={unversionedId:"scn_dataset",id:"scn_dataset",isDocsHomePage:!1,title:"scNavigator dataset",description:"Single-cell Navigator requires several files to be present for each dataset:",source:"@site/docs/dataset.md",slug:"/",permalink:"/scn/docs/docs/",editUrl:"https://github.com/ctlab/scNavigator/edit/master/scn_docs/docs/dataset.md",version:"current",sidebar:"docs",next:{title:"scNavigator dataset descriptor",permalink:"/scn/docs/docs/scn_dataset_json"}},c=[],s={toc:c};function u(e){var t=e.components,n=Object(a.a)(e,["components"]);return Object(i.b)("wrapper",Object(r.a)({},s,n,{components:t,mdxType:"MDXLayout"}),Object(i.b)("p",null,"Single-cell Navigator requires several files to be present for each dataset:"),Object(i.b)("p",null,"General (these files are required):"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"dataset.json")," (required) contains description of the dataset."),Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"plot_data.json")," (required) contains calculated information about every cell (like clustering and tSNE coordinates) as well as precalculated annotations (like cluster borders)")),Object(i.b)("p",null,"Expression data:"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"exp_data.json")," contains gene names and cell barcodes in the same order as they appear in expression matrix, as well as number of total UMIs in the cell."),Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"data.h5")," is an expression (count) matrix. HDF5 allows storing counts effectively: since for the scNavigator we mostly need to look expression of a gene in the datasets, HDF5 can effectively compress columns of integers.")),Object(i.b)("p",null,"Markers:"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"markers.json")," json file describing gene expression markers (optional)")),Object(i.b)("p",null,"Gene signature search:"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"species.modules.gmt")," where ",Object(i.b)("inlineCode",{parentName:"li"},"species")," are ",Object(i.b)("inlineCode",{parentName:"li"},"hs"),", ",Object(i.b)("inlineCode",{parentName:"li"},"mm")," or ",Object(i.b)("inlineCode",{parentName:"li"},"rn")," depending on your dataset"),Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"modules.annotation.json")," describing each of the modules present in the ",Object(i.b)("inlineCode",{parentName:"li"},"species.modules.gmt"))),Object(i.b)("p",null,"Extra files:"),Object(i.b)("ul",null,Object(i.b)("li",{parentName:"ul"},Object(i.b)("inlineCode",{parentName:"li"},"files")," is a directory (next to ",Object(i.b)("inlineCode",{parentName:"li"},"dataset.json"),") where you can put any additional files of choice (optional)")),Object(i.b)("p",null,"In the next vignettes we will describe structure of every of these files."))}u.isMDXComponent=!0},87:function(e,t,n){"use strict";n.d(t,"a",(function(){return p})),n.d(t,"b",(function(){return m}));var r=n(0),a=n.n(r);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function l(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function c(e,t){if(null==e)return{};var n,r,a=function(e,t){if(null==e)return{};var n,r,a={},i=Object.keys(e);for(r=0;r<i.length;r++)n=i[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(r=0;r<i.length;r++)n=i[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var s=a.a.createContext({}),u=function(e){var t=a.a.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):l(l({},t),e)),n},p=function(e){var t=u(e.components);return a.a.createElement(s.Provider,{value:t},e.children)},b={inlineCode:"code",wrapper:function(e){var t=e.children;return a.a.createElement(a.a.Fragment,{},t)}},d=a.a.forwardRef((function(e,t){var n=e.components,r=e.mdxType,i=e.originalType,o=e.parentName,s=c(e,["components","mdxType","originalType","parentName"]),p=u(n),d=r,m=p["".concat(o,".").concat(d)]||p[d]||b[d]||i;return n?a.a.createElement(m,l(l({ref:t},s),{},{components:n})):a.a.createElement(m,l({ref:t},s))}));function m(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=n.length,o=new Array(i);o[0]=d;var l={};for(var c in t)hasOwnProperty.call(t,c)&&(l[c]=t[c]);l.originalType=e,l.mdxType="string"==typeof e?e:r,o[1]=l;for(var s=2;s<i;s++)o[s]=n[s];return a.a.createElement.apply(null,o)}return a.a.createElement.apply(null,n)}d.displayName="MDXCreateElement"}}]);